### ISSUE TITLE
Electrum daemon perf issue: Sync time increasing even wallets are removed

### ISSUE THREAD

**Issue by santzi** [spesmilo/electrum#5613]
Used Electrum version is 3.3.6. I haven't use the latest version yet.
I'm using mainly Electrum Client via terminal and I have noticed issue with wallet synchronization.
My wallets are quite big, about 500 addresses and ElectrumX server running on localhost.

Steps:
1. Add a new wallet (load_wallet). Every wallet has different content.
2. Check status ("electrum daemon status") Sync is "True" state very fast. About 1-2sec
3. Add four wallets more and check status. Sync time increasing and increasing all the time...might be over 15secs easily. (this might be the first bug?)
4. Then close all other wallets (close_wallet) except the latest one.
5. You have now one wallet left. Add a new wallet (load_wallet) and sync time might be over 30 seconds. (perf issue?)
6. stop daemon and start it again. Add two different wallets and sync time is couple of seconds.

Wallet file format (only addresses and imported):
'{
    "addresses": {
        "1AYwGPvUiwYjSL4F8cvnYn4pPFAPzpeLae": {},
        "1BwWb7jM2of2kLhgCEpb7zR8ffMKyfqZ3P": {},
        "17MYxo8qREqWD3QVBqHtiCg3neL7HYnJfK": {},
        "175Q12DS9LTGFYNvjvfBfGpi5t5fauBKGN": {},
        "1KabzT9CrAWxc7tWhCmJfHrC4UjRtTQMgv": {},
        "1Fex8Zt8oxZMM41fMqxZJH4YnnNRFPCr4y": {},
        "1sbDmZ41zD4RETvH1DtS2JhSCVAr69LPE": {},
        "15H7wLeDHKr83G2PcAb4vRMCFbAwye2ZED": {},
        ...
    },
    "seed_version": 18,
    "spent_outpoints": {},
    "stored_height": 593167,
    "transactions": {},
    "tx_fees": {},
    "txi": {},
    "txo": {},
    "use_encryption": false,
    "verified_tx3": {},
    "wallet_type": "imported",
    "winpos-qt": [
        250,
        150,
        840,
        407
    ]
}

**Comment by ecdsa** [comment]
your electrum version is too old, please update before you post here

**Comment by santzi** [comment]
Tested also with version 3.3.8 and the issue is still exist.
Added some measurements (file electrum_client_perf_log.txt attached)

[electrum_client_perf_log.txt](https://github.com/spesmilo/electrum/files/3583455/electrum_client_perf_log.txt)

**Comment by ecdsa** [comment]
could you try on master?
you need to run 
```
electrum daemon -d
electrum load_wallet
electrum list_wallets
```

**Comment by santzi** [comment]
I tried..installation was ok, but after that I got issue.
libsecp256k1 module missing even pip3 says it's installed OK, but python3 import failed.
I don't know how to fix this. Installed 3.3.8 back and it working without this issue.

LinuxUser@server:~$ electrum version
Traceback (most recent call last):
  File "/usr/local/bin/electrum", line 4, in <module>
    __import__('pkg_resources').run_script('Electrum==3.3.8', 'electrum')
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 658, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1438, in run_script
    exec(code, namespace, namespace)
  File "/usr/local/lib/python3.6/dist-packages/Electrum-3.3.8-py3.6.egg/EGG-INFO/scripts/electrum", line 80, in <module>
    from electrum.logging import get_logger, configure_logging
  File "/usr/local/lib/python3.6/dist-packages/Electrum-3.3.8-py3.6.egg/electrum/__init__.py", line 3, in <module>
    from .wallet import Wallet
  File "/usr/local/lib/python3.6/dist-packages/Electrum-3.3.8-py3.6.egg/electrum/wallet.py", line 50, in <module>
    from .simple_config import get_config
  File "/usr/local/lib/python3.6/dist-packages/Electrum-3.3.8-py3.6.egg/electrum/simple_config.py", line 13, in <module>
    from . import constants
  File "/usr/local/lib/python3.6/dist-packages/Electrum-3.3.8-py3.6.egg/electrum/constants.py", line 30, in <module>
    from . import bitcoin
  File "/usr/local/lib/python3.6/dist-packages/Electrum-3.3.8-py3.6.egg/electrum/bitcoin.py", line 34, in <module>
    from . import ecc
  File "/usr/local/lib/python3.6/dist-packages/Electrum-3.3.8-py3.6.egg/electrum/ecc.py", line 45, in <module>
    do_monkey_patching_of_python_ecdsa_internals_with_libsecp256k1()
  File "/usr/local/lib/python3.6/dist-packages/Electrum-3.3.8-py3.6.egg/electrum/ecc_fast.py", line 189, in do_monkey_patching_of_python_ecdsa_internals_with_libsecp256k1
    raise Exception('libsecp256k1 library not available. '
Exception: libsecp256k1 library not available. Verifying Lightning channels is too computationally expensive without libsecp256k1, aborting.
LinuxUser@server:~$ pip3 install libsecp256k1
Collecting libsecp256k1
Installing collected packages: libsecp256k1
Successfully installed libsecp256k1-0.1

LinuxUser@server:~$ python3
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import libsecp256k1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'libsecp256k1'
>>> 
[2]+  Stopped                 python3
LinuxUser@server:~$ pip3 install libsecp256k1
Collecting libsecp256k1
Installing collected packages: libsecp256k1
Successfully installed libsecp256k1-0.1

After this libsecp256k1 installation, issue is still exist.

Should I create a new issue about this?

**Comment by SomberNight** [comment]
> Should I create a new issue about this?

No, it's being tracked in https://github.com/spesmilo/electrum/issues/5606

Try with https://github.com/spesmilo/electrum/commit/251db638af2f9c892a41af941b587729ba1c6771

**Comment by santzi** [comment]
Master 6-Sep compiled ok with 251db63, but I got a new issue.
I added a new wallet, which has couple of addresses. This is imported (only addresses and view only) using this same Master version via GUI (sync ok and show balance zero --> ok).
Then I tried this same wallet use with CLI Daemon.
load_wallet --> ok and synchronized is true
getbalance --> resp "wallet not loaded", should be zero
 is_synchronized --> resp "wallet not loaded", should be TRUE

This is the used wallet:
```
{
    "addr_history": {},
    "addresses": {
        "1MnjTmtpgcC6h7WrYHZVEbRb8HJA9UQrC7": {},
        "14CtVZDRk3QLdyzaouRhxFVk9xCw8fyobJ": {},
        "14ZiSTCoVntEJK9BEcPqSq1yWXXnT5XF88": {},
        "127SNJ1RQsK4Bdz8LfFDMdrRUiixdhR8Ds": {},
        "1yxrZPfbDVGLKb6SE3eckGJ7jEr4k3Ntb": {},
        "1HXjrJU1kPLtu51SSVAav2VbaLugf2C2Uy": {},
        "16XDmsvaxtu5SvZVxEtUinthbtMYbP7ugq": {}
    },
    "qt-console-history": [],
    "seed_version": 18,
    "spent_outpoints": {},
    "stored_height": 591916,
    "transactions": {},
    "tx_fees": {},
    "txi": {},
    "txo": {},
    "use_encryption": false,
    "verified_tx3": {},
    "wallet_type": "imported",
    "winpos-qt": [
        5,
        42,
        840,
        432
    ]
}
```

This console output:
```
LinuxUser@server:/media/tmp$ electrum daemon -d
starting daemon (PID 10689)
LinuxUser@server:/media/tmp$ electrum list_wallets
[]
LinuxUser@server:/media/tmp$ electrum getinfo
{
    "auto_connect": true,
    "blockchain_height": 593958,
    "connected": true,
    "default_wallet": "/home/LinuxUser/.electrum/wallets/default_wallet",
    "fee_per_kb": 4121,
    "path": "/home/LinuxUser/.electrum",
    "server": "192.168.56.1",
    "server_height": 593958,
    "spv_nodes": 1,
    "version": "3.3.8"
}

LinuxUser@server:/media/tmp$ electrum load_wallet -w TestWallet
true
LinuxUser@server:/media/tmp$ electrum list_wallets
[
    {
        "path": "/media/tmp/TestWallet",
        "synchronized": true
    }
]

LinuxUser@server:/media/tmp$ electrum getbalance -w /media/tmp/TestWallet
wallet not loaded
LinuxUser@server:/media/tmp$ electrum is_synchronized -w /media/tmp/TestWallet
wallet not loaded
LinuxUser@server:/media/tmp$ electrum list_wallets
[
    {
        "path": "/media/tmp/TestWallet",
        "synchronized": true
    }
]
LinuxUser@server:/media/tmp$ electrum getinfo
{
    "auto_connect": true,
    "blockchain_height": 593958,
    "connected": true,
    "default_wallet": "/home/LinuxUser/.electrum/wallets/default_wallet",
    "fee_per_kb": 4036,
    "path": "/home/LinuxUser/.electrum",
    "server": "192.168.56.1",
    "server_height": 593958,
    "spv_nodes": 1,
    "version": "3.3.8"
}

```

**Comment by ecdsa** [comment]
your latest issue is fixed in 103a37b0ca597be27760ef7c206c9f724fc6ed3b
